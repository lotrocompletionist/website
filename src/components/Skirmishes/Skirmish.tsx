import * as React from "react";
import {
  CardContent,
  Media,
  MediaLeft,
  MediaContent,
  Content,
  Image,
  Title,
  Subtitle
} from "bloomer";
import { LevelRange } from "@app/components/Shared/LevelRange/LevelRange";
import { Bosses } from "@app/components/Shared/Bosses/Bosses";
import { ISkirmish } from "@app/data/skirmish";

export interface SkirmishProps {
  skirmish: ISkirmish;
}

export function Skirmish(props: SkirmishProps) {
  return (
    <>
      <CardContent>
        <Media>
          <MediaLeft>
            <Image isSize="48x48" src="https://via.placeholder.com/96x96" />
          </MediaLeft>
          <MediaContent>
            <Title isSize={4}>{props.skirmish.name}</Title>
          </MediaContent>
        </Media>
        <Content>
          {props.skirmish.type}
          <br />
          <small>
            <LevelRange range={props.skirmish.level} />
            <br />
            <Bosses bosses={props.skirmish.bosses} />
          </small>
        </Content>
      </CardContent>
    </>
  );
}
