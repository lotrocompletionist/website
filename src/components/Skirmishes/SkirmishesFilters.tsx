import * as React from "react";
import { factions, types } from "@app/data/skirmishes";
import { DropdownFilter } from "@app/components/Shared/Filtering/DropdownFilter";
import { DropdownCheckboxFilterItem } from "@app/components/Shared/Filtering/DropdownFilterItem";

interface SkirmishesFiltersProps {
  onFactionChange: React.ReactEventHandler<HTMLInputElement>;
  onTypeChange: React.ReactEventHandler<HTMLInputElement>;
}

export function SkirmishesFilters(props: SkirmishesFiltersProps) {
  return (
    <>
      <DropdownFilter
        filterItem={DropdownCheckboxFilterItem}
        header="Faction"
        values={factions}
        onChange={props.onFactionChange}
      />
      <DropdownFilter
        filterItem={DropdownCheckboxFilterItem}
        header="Type"
        values={types}
        onChange={props.onTypeChange}
      />
    </>
  );
}
