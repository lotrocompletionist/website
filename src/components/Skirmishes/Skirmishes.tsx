import * as React from "react";
import { skirmishes } from "../../data/skirmish";
import { Skirmish } from "../Skirmish/Skirmish";

export default function Skirmishes() {
  return (
    <>
      {skirmishes.map(skirmish => (
        <Skirmish key={skirmish.name} skirmish={skirmish} />
      ))}
    </>
  );
}
