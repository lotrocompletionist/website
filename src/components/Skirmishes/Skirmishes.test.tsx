import * as React from "react";
import { create } from "react-test-renderer";
import { default as Skirmishes } from "./Skirmishes";

test("<Skirmishes> renders correctly", () => {
  const tree = create(<Skirmishes />).toJSON();
  expect(tree).toMatchSnapshot();
});
