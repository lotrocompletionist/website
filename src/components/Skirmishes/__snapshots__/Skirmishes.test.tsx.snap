// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skirmishes> renders correctly 1`] = `
Array [
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          The Siege of Gondamon
        </h1>
      </div>
    </article>
    <div
      className="content"
    >
      Defense
      <br />
      <small>
        Level: 
        20
         - 115
        <br />
        Bosses: 
        Thrastar, Hergill, Agi Gatebreaker, Ghashnâkh, Sigurmar
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          Trouble in Tuckborough
        </h1>
      </div>
    </article>
    <div
      className="content"
    >
      Offense
      <br />
      <small>
        Level: 
        20
         - 115
        <br />
        Bosses: 
        Ghâshfog, Murbarash, Fíthrokh, Oldroot, Big Fright
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          Storm on Methedras
        </h1>
      </div>
    </article>
    <div
      className="content"
    >
      Offense
      <br />
      <small>
        Level: 
        20
         - 115
        <br />
        Bosses: 
        Rhodri, Budant, Maru Brenin, Garzog, Caur, Helechúl, Núrvor, Dombor, Daeharn
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          Stand at Amon Sûl
        </h1>
      </div>
    </article>
    <div
      className="content"
    >
      Defense
      <br />
      <small>
        Level: 
        25
         - 115
        <br />
        Bosses: 
        Golnauk, Bloodwing, Gúgaw's Arm, Laimbenn, Mokum, Gwandir, Norgol, Gwathrengor, Urkflagít
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          Attack at Dawn
        </h1>
      </div>
    </article>
    <div
      className="content"
    >
      Offense
      <br />
      <small>
        Level: 
        30
         - 115
        <br />
        Bosses: 
        Tharb, Pugh, Fríb, Zímarp, Krampum, Kaushar
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          Thievery and Mischief
        </h1>
      </div>
    </article>
    <div
      className="content"
    >
      Offense
      <br />
      <small>
        Level: 
        30
         - 115
        <br />
        Bosses: 
        Lordfisher, Padfoot, Goretusk, Ned Ashleaf, Tom Greythistle, Túkar, Snagbúrz, Gruglup
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          Defence of the Prancing Pony
        </h1>
      </div>
    </article>
    <div
      className="content"
    >
      Defense
      <br />
      <small>
        Level: 
        35
         - 115
        <br />
        Bosses: 
        Southpaw, Old Hunter, Ufglup, Rat-catcher, Mukghashan, Kamhont, Banosnákh, Willie Hawthorn
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          The Ford of Bruinen
        </h1>
      </div>
    </article>
    <div
      className="content"
    >
      Defense
      <br />
      <small>
        Level: 
        40
         - 115
        <br />
        Bosses: 
        Zanúrz, Kunquin, Durkúf, Bruidagnir
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          The Icy Crevasse
        </h1>
      </div>
    </article>
    <div
      className="content"
    >
      Offense
      <br />
      <small>
        Level: 
        40
         - 115
        <br />
        Bosses: 
        Míthalagos, Kókas-joku, Lapa-kita, Coflúg, Crantok, Túgarch, Veri-torvi, Mahti
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          Barrow-downs Survival
        </h1>
      </div>
    </article>
    <div
      className="content"
    >
      Survival
      <br />
      <small>
        Level: 
        45
         - 115
        <br />
        Bosses: 
        
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          The Battle of the Deep-way
        </h1>
      </div>
    </article>
    <div
      className="content"
    >
      Defense
      <br />
      <small>
        Level: 
        55
         - 115
        <br />
        Bosses: 
        Lhenathronil, Gamurz-ash, Earthbreaker, Forktongue, Foulscum, Gorged Queen, Burntail, Mad Croaker
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          Battle of the Way of Smiths
        </h1>
      </div>
    </article>
    <div
      className="content"
    >
      Defense
      <br />
      <small>
        Level: 
        55
         - 115
        <br />
        Bosses: 
        Fire-heart, Ice-heart, Shadow-heart, Celebarch, Fikûhom, Durbrennil, Sadâuk
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          Battle of the Twenty-first Hall
        </h1>
      </div>
    </article>
    <div
      className="content"
    >
      Defense
      <br />
      <small>
        Level: 
        55
         - 115
        <br />
        Bosses: 
        Morûrz-kala, Kalus, Throqûm, Fodhroval, Shadow from the Heart of the World, Helchnaeth
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          Strike Against Dannenglor
        </h1>
      </div>
    </article>
    <div
      className="content"
    >
      Offense
      <br />
      <small>
        Level: 
        60
         - 115
        <br />
        Bosses: 
        Helchfaer, Ghostfang, Ruingalad, Swartclaw, Rimdeloth, Gorogmúl, Gothling, Fiklúk, Seregruin
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          Protectors of Thangúlhad
        </h1>
      </div>
    </article>
    <div
      className="content"
    >
      Defense
      <br />
      <small>
        Level: 
        60
         - 115
        <br />
        Bosses: 
        Borndol, Ruithroval, Azrí, Hokarul, Grudom, Frúzglob, Golugthrug, Lapus, Torúglup
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          Breaching the Necromancer's Gate
        </h1>
      </div>
    </article>
    <div
      className="content"
    >
      Offense
      <br />
      <small>
        Level: 
        65
         - 115
        <br />
        Bosses: 
        Lavalúg, Gortag, Firingúl, Murúk, Annoniel, Thrângund, Morgúr, Agardúr, Grishglok
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          Assault on the Ringwraiths' Lair
        </h1>
      </div>
    </article>
    <div
      className="content"
    >
      Offense
      <br />
      <small>
        Level: 
        65
         - 115
        <br />
        Bosses: 
        Faelathron, Mazaukgrat, Kúfit, Durnákh, Gúlheryn, Afglu, Zánurz, Norchador, Mornákh
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          The Battle in the Tower
        </h1>
      </div>
    </article>
    <div
      className="content"
    >
      Offense
      <br />
      <small>
        Level: 
        65
         - 115
        <br />
        Bosses: 
        Thangaran, Iaerien, Hogaward, Múdku, Blord, Gundnákh, Gadhumúrz, Urbor
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          Rescue in Nûrz Ghâshu
        </h1>
      </div>
    </article>
    <div
      className="content"
    >
      Offense
      <br />
      <small>
        Level: 
        65
         - 115
        <br />
        Bosses: 
        Cefrog, Gháshglob, Gormalith, Lúgryn, Morfuin, Narubras, Nurnor, Snagtalún
      </small>
    </div>
  </div>,
]
`;
